dspace-server-webapp/src/main/java/org/dspace/app/rest/model/ProcessFileWrapperRest.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.model;
+
+import java.util.List;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+
+
+/**
+ * The ProcessFileWrapper REST Resource
+ *
+ */
+public class ProcessFileWrapperRest implements RestModel {
```

Comment:
This one is needed for the following endpoint: https://github.com/DSpace/Rest7Contract/pull/99/files#diff-12aac6d84a3393b361553d3f2fe93962R141, we need a way here to have in our output:
* A process identifier
* A list of files

This object provides that wrapper, we can always rename it if you wish.

PS: JavaDocs have been added.

Reply:
@KevinVdV : OK, I see. It seems like it's more about grouping this files by *type*.  So, we might consider calling this the `ProcessFileTypeWrapperRest`.  But, I don't feel strongly either way here.

One thing to note though is that the behavior of this endpoint is a bit different than normal, so we might want to verify with the Angular team that caching will work appropriately here.  Essentially, this endpoint is returning `ProcessFileWrapperResource` which is just a wrapper around a `BitstreamResource`.  So, the client side / UI needs to be aware of that so that it does **NOT** cache the `ProcessFileWrapper` separate from cached `Bitstream` objects (as treating these separately could end up duplicating some caching).  

@artlowel : have you seen the approach of this PR?

Normally, to make this similar to other existing endpoints, the behavior here would be to have the ``/processes/<:process-id>/files` endpoint directly return `BitstreamResource` objects.  But, that'd require rethinking the REST Contract slightly (as this `/files` path could no longer be grouped by `dspace.process.filetype`....but we still could have `/files/[:type]` return just the file of that type).

I don't feel strongly either way here. I'm just pointing out that this endpoint is different from how others normally behave.  If @artlowel sees no issues with caching on the Angular side, then I'm OK with this approach.  Otherwise, if needed, we could rethink the Contract again to align this endpoint more with others that act similarly (e.g. `/bundles/[:id]/bitstreams` is somewhat similar in that it returns just a list of bitstreams, in that case belonging to a specific bundle)

Reply:
Thanks for the heads up @tdonohue. That /files endpoint as described in the rest contract is indeed problematic. I'll discuss it with Kevin and Ben to see if we can't come up with something better.

Reply:
@tdonohue In discussion with Art & Ben I created the following contract PR: https://github.com/DSpace/Rest7Contract/pull/130, feel free to add feedback there, once accepted, we will implement these changes.

Reply:
@KevinVdV : I've added a review to the contract changes. They look good overall, I just added minor rewording suggestions.

Based on the new contract changes, all my feedback in this discussion thread above can be ignored, and I'd expect **all** of the `ProcessFileWrapper*` classes to now be removed from this PR.  We should be able to now use a simple `LinkRepository` to return the Bitstreams associated with the Process.